satpy/resample.py

```diff
@@ -1046,6 +1050,26 @@ def compute(self, data, expand=True, **kwargs):
         return update_resampled_coords(data, new_data, target_geo_def)
 
 
+def _get_arg_to_pass_for_skipna_handling(**kwargs):
+    """Determine if skipna can be passed to the compute functions for the average and sum bucket resampler."""
+    # FIXME this can be removed once Pyresample 1.18.0 is a Satpy requirement
+
+    if PR_USE_SKIPNA:
+        if 'mask_all_nan' in kwargs:
+            warnings.warn('Argument mask_all_nan is deprecated. Please use skipna for missing values handling. '
+                          'Continuing with default skipna=True, if not provided differently.', DeprecationWarning)
+            kwargs.pop('mask_all_nan')
+    else:
+        if 'mask_all_nan' in kwargs:
+            warnings.warn('Argument mask_all_nan is deprecated.'
+                          'Please update Pyresample and use skipna for missing values handling.',
+                          DeprecationWarning)
+        kwargs.setdefault('mask_all_nan', False)
+        kwargs.pop('skipna')
```

Comment:
Is skipna always provided? Otherwise, you might need a `, None)` at the end of the pop. Otherwise I think it raises a KeyError if it is missing.

Reply:
Yes, it's always provided, so  it can stay like this.